File: change_dhcp_to_static.py
#!/usr/bin/python
import glob
import os;

def get_host_ip():
    import commands
    cmd_line = commands.getoutput ('hostname -I ')
    output=cmd_line.split();
    ip= filter(lambda x:  x.startswith("192.168") , output); 
    assert(len(ip) !=0);
    return ip;

def is_static(if_cfg):
    import re;
    all_the_text = open(if_cfg).read( )     
    pattern=re.compile('BOOTPROTO.*static');
    match=pattern.search(all_the_text)
    if match: 
        return 1;
    return None;

def append_static_lines(cfg_file, ip):
    os.system("sed 's/ONBOOT=no/#ONBOOT=no/g'  -i  %s "%cfg_file );

    fh = open(cfg_file, 'a')
    string="""
ONBOOT=yes
BOOTPROTO=static
IPADDR=%s
NETMASK=255.255.255.0
GATEWAY=192.168.1.1
DNS=8.8.8.8
NM_CONTROLLED="no"
"""%(ip);
    fh.write("%s\n"%( string) )
    fh.close(); 


for file in  filter(lambda x: x!=None and not x.endswith("lo"), map (lambda x: x  if os.path.isfile(x) else None, glob.glob("/etc/sysconfig/network-scripts/ifcfg-*"))):
    print file;
    if  is_static(file):
        print "STATIC: %s skip now "%file;
        next;
    ip=get_host_ip()[0];
    append_static_lines(file,  ip);

File: change_dhcp_to_static.pl
#!/usr/bin/perl
#centos7 dhcpd change to static 


for $each (glob("/etc/sysconfig/network-scripts/ifcfg-*"))
{
    print $each."\n";;
    if($each=~/lo$/)
    {next;}

    if (is_static($each))
    {
        print "STATIC:".$each.", skip now \n";
        next;
    }

    my $ip=get_host_ip();
    append_static_lines($each,$ip );
}

########################################################################
# /etc/sysconfig/network-scripts/ifcfg-eno1
sub  is_static($)
{
    (my $cfg_file)=@_;
    open(FILE, $cfg_file)  or die("open $cfg_file error \n");
    for(<FILE>)
    {
        if($_=~/BOOTPROTO.*static/)
        {
            close(FILE);
            return 1; 
        }
    }
    close(FILE);
    return undef;
}

#==========================================================================
sub  get_host_ip()
{
    $buffer=`hostname -I `;
    @array=grep {"192.168."}  split(/\s+/, $buffer);
    return $array[0];
}

#==========================================================================
sub append_static_lines($$)
{
    (my $cfg_file, my $ip )=@_;
    system("sed 's/ONBOOT=no/#ONBOOT=no/g'  -i  $cfg_file " );

    open(APPEND, ">>$cfg_file") or die("open $cfg_file append error \n");
    print APPEND <<EOF

ONBOOT=yes
BOOTPROTO=static
IPADDR=$ip
NETMASK=255.255.255.0
GATEWAY=192.168.1.1
DNS=8.8.8.8
NM_CONTROLLED="no"

EOF
;

}
